modules/manifold/src/reducers/index.js

```diff
@@ -1,101 +1,119 @@
 import {handleActions} from 'redux-actions';
 import reduceReducers from 'reduce-reducers';
 import keplerGlReducer from 'kepler.gl/reducers';
+
+import {handleUpdateMetric} from './data-generation';
+import {
+  handleUpdateSegmentationMethod,
+  handleUpdateBaseCols,
+  handleUpdateNClusters,
+  handleUpdateSegmentFilters,
+  handleUpdateSegmentGroups,
+} from './data-slicing';
+
 import {
-  UPDATE_DIVERGENCE_THRESHOLD,
-  FETCH_BACKEND_DATA_START,
-  FETCH_BACKEND_DATA_SUCCESS,
   FETCH_MODELS_START,
   FETCH_MODELS_SUCCESS,
+  FETCH_BACKEND_DATA_START,
+  FETCH_BACKEND_DATA_SUCCESS,
   FETCH_FEATURES_START,
   FETCH_FEATURES_SUCCESS,
   LOAD_LOCAL_DATA_START,
   LOAD_LOCAL_DATA_SUCCESS,
   LOAD_LOCAL_DATA_FAILURE,
-  UPDATE_FEATURE_TYPES,
   UPDATE_SELECTED_INSTANCES,
+  UPDATE_DIVERGENCE_THRESHOLD,
   UPDATE_SELECTED_MODELS,
   UPDATE_N_CLUSTERS,
   UPDATE_METRIC,
   UPDATE_SEGMENTATION_METHOD,
   UPDATE_SEGMENT_FILTERS,
-  UPDATE_BASE_MODELS,
+  UPDATE_BASE_COLS,
   UPDATE_SEGMENT_GROUPS,
   UPDATE_DISPLAY_GEO_FEATURES,
   UPDATE_COLOR_BY_FEATURE,
-} from './actions';
-import {DEFAULT_FEATURE_TYPES} from './constants';
-import {getDefaultSegmentGroups, registerExternalReducers} from './utils';
+} from '../actions';
+
+import {METRIC} from '../constants';
+import {registerExternalReducers} from '../utils';
 
 export const DEFAULT_STATE = {
   // TODO: these are fields used with Python backend. Consider consolidate/remove
   metaData: undefined,
   models: undefined,
   features: undefined,
 
-  // data and metadata
+  /** data and metadata */
   // columns: array of array of columns; fields: array of field metadata
   data: {columns: [], fields: []},
-  // collumnTypeRanges: map from "column type" (x, yPred, etc) to array of 2 elements indicating the start and end index of that column type in dataset
+  // map from "column type" (x, yPred, etc) to array of 2 elements indicating the start and end index of that column type in dataset
```

Comment:
if we use `start` and `end` indices here, are we assuming the columns of the same type will be provided in consecutive order? or we will sort the columns for the user?

Reply:
Yes, we are assuming the columns of the same type will be provided in consecutive order. We don't remove columns (only replace or append) since we refer to columns by ids, and want to ensure the reference is still valid after replacing or appending.